pkg/controller/controller_instance.go

```diff
@@ -1545,10 +1545,15 @@ func (c *controller) prepareServiceInstanceLastOperationRequest(instance *v1beta
 		return nil, err
 	}
 
+	planID := ""
+	if instance.Status.InProgressProperties != nil {
```

Comment:
I'm not sure we're allowed to ever call the broker with an empty `PlanID` (according to the OSBAPI spec), so I'm thinking the controller should instead fail fast here. `InProgressProperties` should never be nil at this point, so I'd treat it being nil as an error and log it. If there's ever a situation that causes this to be nil, we'd be able to identify the problem fast instead of covering it up.

Reply:
The plan is optional on an [update](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#updating-a-service-instance).  I do agree that with the other changes in this, InProgressProperties should never be nil  now so I've updated to fail with an error if it is.  Thanks for the good input @luksa